### PR TITLE
Const trickle fix as discovered with discord connector issue

### LINKED ISSUE #749: `use` in trickle has problems with constants
**Problem**

There are two issues with included const in trickle:

1) consts are not found in selects: https://github.com/tremor-rs/tremor-runtime/blob/c814b4b2bfd9a5dba4f128f064f54ec70d1b1e57/tests/queries/pp_const
2) consts used in functions that are included fail: https://github.com/tremor-rs/tremor-runtime/blob/c814b4b2bfd9a5dba4f128f064f54ec70d1b1e57/tests/queries/pp_fn

**Possible Solution(s)**
We should allow consts with include in trickle

### PR DESCRIPTION
**Pull request by Licenser**
# Pull request

## Description

This adds a discord linked connector to allow writing bots in tremor. In addition, it fixes some bugs discovered during the exercise.

## Related

* Related Issues: fixes #749
* Related [docs PR](https://github.com/tremor-rs/tremor-www-docs/pull/77)

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes or other observable changes in behavior

### CODE REVIEW

**Review by darach — APPROVED** [review]
LGTM 👍

File: src/codec/binflux.rs [hunk 1]
```diff
@@ -42,18 +42,18 @@ impl BInflux {
 
         let mut res = Vec::with_capacity(512);
         res.write_u16::<BigEndian>(0)?;
-        if let Some(measurement) = v.get("measurement").and_then(Value::as_str) {
+        if let Some(measurement) = v.get_str("measurement") {
```
Comment (mfelsche) [inline src/codec/binflux.rs:45]: aaahhh, so good!

Comment (Licenser) [reply src/codec/binflux.rs:45]: Ja, I had to write so many of those `and_then` that I realized that is something missing in the value-trait.

Comment (darach) [reply src/codec/binflux.rs:45]: 💅  That is sweetness itself!

File: src/source/discord.rs [hunk 2]
```diff
@@ -0,0 +1,873 @@
...
+            };
+
+            Ok(Box::new(Self {
+                origin_uri,
+                config,
+                onramp_id: id.clone(),
+                client: None,
+            }))
+        } else {
+            Err("Missing config for crononome onramp".into())
```
Comment (mfelsche) [inline src/source/discord.rs:146]: misleading error message

File: tests/script_errors/double_const/script.tremor [hunk 3]
```diff
@@ -1,2 +1,2 @@
 const one = 1;
-const one = two;
+const one = "two";
```
Comment (mfelsche) [inline tests/script_errors/double_const/script.tremor:2]: what about conflicting names with other entities like functions/locals?

**Pull request merged**